File: code/c/array/test-for.c
/* Program: test-for.c */
#include <stdio.h>

void print_characters(const char *text, int size)
{
    int i;
    for (i = 0; i < size; i++)
    {
        printf("%c (ASCII %hd) at index %d\n", text[i], text[i], i);
    }
}

int main()
{
    print_characters("Hello World", 12);
    print_characters("Fred", 5);
    return 0;
}

File: code/pascal/array/TestFor.pas
program TestFor;

procedure PrintLine(length: Integer);
var
  i : Integer;
begin
    for i := 1 to length do Write('-');
    WriteLn();
end;

procedure PrintCharacters(text: String);
var
  i : Integer;
begin
    for i := 1 to Length(text) do
    begin
        WriteLn(text[i], ' (ASCII ', Integer(text[i]), ') at index ', i);
    end;
end;

begin
  PrintCharacters('Hello World');
  PrintLine(50);
  PrintCharacters('Fred');
end.
